cogs/moderation.py

```diff
@@ -97,7 +97,54 @@ async def find(self, ctx, member: utils.MemberLookupConverter):
         else:
             embed.add_field(name="Punishments:", value="No punishment or blacklist logs found!")
             await ctx.send(embed=embed)
+            
+    @commands.command(usage='changename <member> <newname>', description="Change the users name.")
+    @commands.guild_only()
+    @checks.is_security_or_higher_check()
+    async def changename(self, ctx, member: utils.MemberLookupConverter, newname):
+        async with aiohttp.ClientSession() as cs:
+            async with cs.get(f'https://rotmg-discord-bot.wm.r.appspot.com/?player={newname}', ssl=False) as r:
+                if r.status == 403:
+                    print("ERROR: API ACCESS FORBIDDEN")
+                    await ctx.send(f"<@{self.client.owner_id}> ERROR: API ACCESS REVOKED!.")
+                data = await r.json()  # returns dict
+        if not data:
+            return await ctx.send("There was an issue retrieving realmeye data. Please try the command later.")
+        if 'error' in data:
+            embed = discord.Embed(title='Error!', description=f"There were no players found on realmeye with the name `{altname}`.",
+                                  color=discord.Color.red())
+            return await ctx.send(embed=embed)
+
+        cleaned_name = str(data["player"])
+
+        res = await sql.change_username(self.client.pool, member.id, cleaned_name)
+        if not res:
+            embed = discord.Embed(title="Error!", description="Something went wrong, username couldn't be changed in SQL", color=discord.Color.red())
+            return await ctx.send(embed=embed)
+
+        embed = ""
+        for g in self.client.guilds:
+            m = await g.get_member(member.id)
+            if m:
+                name = m.display_name
+
+                if cleaned_name.lower() in name.lower():
+                    embed = discord.Embed(title="Error!", description="Name specified is the same name as the user's name currently!",
+                                        color=discord.Color.red())
+                    
+                separator = " | "
+                s_name = name.split(separator)
+                s_name[0] = newname
```

Comment:
Wouldn't you be looking for an index that has the new name rather than the first name?